File: backend/app/infrastructure/repositories/mappers/lesson_mapper.py
from app.domain.entities.lesson import Lesson as LessonDomain
from app.infrastructure.repositories.orm.lesson import Lesson as LessonORM

def to_domain(orm_lesson: LessonORM) -> LessonDomain:
    return LessonDomain(
        id=orm_lesson.id,
        course_id=orm_lesson.course_id,
        title=orm_lesson.title,
        content_type=orm_lesson.content_type,
        content_data=orm_lesson.content_data,
        order=orm_lesson.order
    )

File: frontend/lib/data/mappers/lesson_mapper.dart
// frontend/lib/data/mappers/lesson_mapper.dart
import 'package:frontend/data/models/lesson_model.dart';
import 'package:frontend/domain/entities/lesson.dart';

class LessonMapper {
  static Lesson fromModel(LessonModel model) {
    return Lesson(
      id: model.id,
      courseId: model.courseId,
      title: model.title,
      contentType: model.contentType,
      contentData: model.contentData,
      order: model.order,
    );
  }

  static LessonModel toModel(Lesson entity) {
    return LessonModel(
      id: entity.id,
      courseId: entity.courseId,
      title: entity.title,
      contentType: entity.contentType,
      contentData: entity.contentData,
      order: entity.order,
    );
  }
}
